pkg/randchar/generator.go

```diff
@@ -251,36 +255,48 @@ func (cs Charset) Add(set Charset) Charset {
 		i++
 	}
 
-	return result
+	return Charset{
+		chars: result,
+	}
 }
 
 // Subtract removes all characters from a set that match a given set of characters.
-func (cs Charset) Subtract(set Charset) Charset {
+func (c Charset) Subtract(set Charset) Charset {
 	filter := make(map[byte]struct{})
-	for _, char := range set {
+	for _, char := range set.chars {
 		filter[char] = struct{}{}
 	}
 
 	result := []byte{}
-	for _, char := range cs {
+	for _, char := range c.chars {
 		_, exists := filter[char]
 		if !exists {
 			result = append(result, char)
 		}
 	}
 
-	return result
+	return Charset{
+		chars: result,
+	}
 }
 
 // IsSubset returns true when the character set is a subset of the given set.
 // When both sets are the same it returns true too.
-func (cs Charset) IsSubset(of Charset) bool {
+func (c Charset) IsSubset(of Charset) bool {
+	if c.Size() == 0 {
+		return true
+	}
+
+	if c.Size() > 0 && of.Size() == 0 {
```

Comment:
These size checks are redundant, why would we add them?

Reply:
I mean that

```Go
if c.Size() == 0 {
	return true
}

if c.Size() > 0 && of.Size() == 0 {
	return false
}
```

can be removed, as it is also covered by the set logic below.

Reply:
Yes, functionally they are the same but this adds a small speed optimization. It's not super big, but we can leave it in. Clarified it by restructuring the `if` statements and with a comment.